org.eclipse.m2e.core/pom.xml

```diff
@@ -50,6 +50,7 @@
               <goal>xpp3-writer</goal>
             </goals>
             <configuration>
+              <outputDirectory>${project.basedir}/src-gen</outputDirectory>
```

Comment:
At this point I'm not 100% certain if the generated .java-files are still compiled and included in the jars during Maven build?

Reply:
I think it's fine. Did you try a local build to verify that?

Reply:
My local build is just running (fetching the dependency for the first time takes awhile).
I will check and report the results.

Reply:
I verified that the classes from the now called `src-gen` folder in m2e.core (for example `org.eclipse.m2e.core.internal.lifecyclemapping.model.io.xpp3.LifecycleMappingMetadataSourceXpp3Reader`) are in `m2e.core.jar` as compiled class files and in `m2.core-sources.jar` as java source files.